src/Functional.CQS.AOP.IoC.PureDI.MetricsCapturing.Tests/AsyncQueryHandlerMetricsCapturingDecoratorForUniversalStrategyTests.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoFixture;
+using AutoFixture.Xunit2;
+using FakeItEasy;
+using Functional.CQS.AOP.CommonTestInfrastructure.DummyObjects;
+using Functional.CQS.AOP.MetricsCapturing;
+using Xunit;
+
+namespace Functional.CQS.AOP.IoC.PureDI.MetricsCapturing.Tests
+{
+	public class AsyncQueryHandlerMetricsCapturingDecoratorForUniversalStrategyTests
+	{
+		[Theory]
+		[AsyncQueryHandlerCompletesSuccessfully]
+		public async Task ShouldCaptureResultAndElapsedTime(
+			AsyncQueryHandlerMetricsCapturingDecoratorForUniversalStrategy<DummyQueryReturnsValueType, DummyQueryReturnsValueTypeResult> sut,
+			IUniversalMetricsCapturingStrategy metricsCapturingStrategy)
+		{
+			var query = new DummyQueryReturnsValueType();
+			await sut.HandleAsync(query);
+
+			A.CallTo(() => metricsCapturingStrategy.OnInvocationStart()).MustHaveHappenedOnceExactly();
+			A.CallTo(() => metricsCapturingStrategy.OnInvocationCompletedSuccessfully(A<TimeSpan>._)).MustHaveHappenedOnceExactly();
+			A.CallTo(() => metricsCapturingStrategy.OnInvocationException(A<Exception>._, A<TimeSpan>._)).MustNotHaveHappened();
+		}
+
+		[Theory]
+		[AsyncQueryHandlerThrowsException]
+		public async Task ShouldCaptureExceptionAndElapsedTime(
```

Comment:
Similar comment about the test name not matching the assertions.

Reply:
Same comment applies to `AsyncQueryHandlerMetricsCapturingDecoratorTests`.

Reply:
In this context, "capture" means "the appropriate values were passed to the right method"